layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableLayoutController.kt

```diff
@@ -0,0 +1,356 @@
+     */
+    private fun addDualLandscapeContent() {
+        if (layoutConfig.isDualLandscapeSingleContainer ||
+            layoutConfig.dualLandscapeSingleLayoutId != View.NO_ID
+        ) {
+            if (layoutConfig.dualLandscapeSingleLayoutId != View.NO_ID) {
+                firstContainer.addContent(layoutConfig.dualLandscapeSingleLayoutId)
+
+                if (!layoutConfig.isDualLandscapeSingleContainer) {
+                    layoutConfig.isDualLandscapeSingleContainer = true
+                }
+            }
+
+            removeHinge()
+            updateDimensionsForSingleScreen()
+        } else {
+            val windowRect =
+                normalizeWindowRect(
+                    foldingFeature?.bounds,
+                    rootView.context.getWindowRect(),
+                    VERTICAL
+                )
+            getScreenRectangles(
+                foldingFeature?.bounds,
+                windowRect
+            )?.let { screenRectList ->
+                addDualScreenContent(
+                    VERTICAL,
+                    screenRectList[0],
+                    screenRectList[1]
+                )
+            }
+        }
+    }
+
+    /**
+     * Contains the logic for:
+     * 1) (dual screen landscape) dual portrait - single container
+     * 2) (dual screen landscape) dual portrait - dual container
+     *
+     * Depending on the configuration,
+     * this function will add the content for the first container or for both containers.
+     */
+    private fun addDualPortraitContent() {
+        if (layoutConfig.isDualPortraitSingleContainer ||
+            layoutConfig.dualPortraitSingleLayoutId != View.NO_ID
+        ) {
+            if (layoutConfig.dualPortraitSingleLayoutId != View.NO_ID) {
+                firstContainer.addContent(layoutConfig.dualPortraitSingleLayoutId)
+
+                if (!layoutConfig.isDualPortraitSingleContainer) {
+                    layoutConfig.isDualPortraitSingleContainer = true
+                }
+            }
+
+            removeHinge()
+            updateDimensionsForSingleScreen()
+        } else {
+            val windowRect =
+                normalizeWindowRect(
+                    foldingFeature?.bounds,
+                    rootView.context.getWindowRect(),
+                    HORIZONTAL
+                )
+            getScreenRectangles(
+                foldingFeature?.bounds,
+                windowRect
+            )?.let { screenRectList ->
+                addDualScreenContent(
+                    HORIZONTAL,
+                    screenRectList[0],
+                    screenRectList[1]
+                )
+            }
+        }
+    }
+
+    /**
+     * Inflate content and add it to both containers.
+     *
+     * @param layoutOrientation The [FoldableLayout] orientation
+     * @param startScreenRect The bounds for the first screen
+     * @param endScreenRect The bounds for the second screen
+     */
```

Comment:
nit, but it would be more correct to say: `first display area` and `second display area` since not all devices have two physical screens.

Reply:
Done